browser/src/control/Control.FormulaBarJSDialog.js

```diff
@@ -23,10 +23,7 @@ class FormulaBar {
 		this.map.on('formulabar', this.onFormulaBar, this);
 		this.map.on('jsdialogupdate', this.onJSUpdate, this);
 		this.map.on('jsdialogaction', this.onJSAction, this);
-
 		this.map.on('doclayerinit', this.onDocLayerInit, this);
-		app.events.on('updatepermission', this.onUpdatePermission.bind(this));
```

Comment:
we don't need that anymore?

Reply:
I don't think so, because if I open document in readonly mode it hides the whole toolbar and formular bar.

Reply:
I don't see a problem mobile view when we change to edit mode after clicking on "edit' button